server/routes/shared_api/autocomplete/autocomplete.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from urllib.parse import urlencode
+
+from flask import Blueprint
+from flask import request
+
+from server.routes.shared_api.autocomplete import helpers
+from server.routes.shared_api.place import findplacedcid
+
+# TODO(gmechali): Add Stat Var search.
+
+# Define blueprint
+bp = Blueprint("autocomplete", __name__, url_prefix='/api')
+
+
+@bp.route('/autocomplete', methods=['GET', 'POST'])
+def autocomplete():
+  """Predicts the user query for location only, using the Google Maps prediction API.
+  Returns:
+      Json object represnting 5 location predictions for the query.
+  """
+  lang = request.args.get('hl')
+  original_query = request.args.get('query')
+  query = original_query
+
+  # Extract subqueries from the user input.
+  queries = helpers.find_queries(query)
+
+  # Send requests to the Google Maps Predictions API.
+  prediction_responses = helpers.predict(queries, lang)
+
+  place_ids = []
+  for prediction in prediction_responses:
+    place_ids.append(prediction["place_id"])
+
+  place_id_to_dcid = []
+  if place_ids:
+    place_id_to_dcid = json.loads(findplacedcid(place_ids).data)
+
+  final_predictions = []
+  # TODO(gmechali): See if we can use typed dataclasses here.
```

Comment:
+1

Reply:
So the reason I didnt do it, is because I didnt think it was worth making its own data class. I could add the Place you added to the response, but then it becomes nested one more layer down. 
It's definitely not a big deal to nest it once more either. 
I was planning on leaving it as is for now - but lmk if you'd prefer I do the data class change rn.

Reply:
id like to get to the point where all flask endpoints have a dataclass for their return value  (and if they're a POST request, for the JSON body). . I think this makes it easier for the frontend to see in one place the shape of the payload. But-can be in a follow up since this response will likely be changing when we add stat var search

Reply:
Sounds good. will make sure to refactor it once we have the SV search plan so we can make a dataclass that fits for both.